fetch-publication/src/main/java/no/unit/nva/publication/fetch/FetchPublicationHandler.java

```diff
@@ -5,105 +5,73 @@
-    public static final String PUBLICATION_NOT_FOUND = "Publication not found.";
+public class FetchPublicationHandler extends ApiGatewayHandler<Void, JsonNode> {
 
     public static final String PUBLICATION_CONTEXT_JSON = "publicationContext.json";
 
-    private final transient PublicationService publicationService;
+    private final PublicationService publicationService;
+    private final ObjectMapper objectMapper;
 
     /**
      * Default constructor for MainHandler.
      */
-    @JacocoGenerated
     public FetchPublicationHandler() {
-        this(PublicationHandler.createObjectMapper(),
-            new RestPublicationService(
+        this(new RestPublicationService(
                     HttpClient.newHttpClient(),
+                    ObjectMapperConfig.objectMapper,
                     new Environment()),
+                ObjectMapperConfig.objectMapper,
                 new Environment());
     }
 
     /**
      * Constructor for MainHandler.
      *
-     * @param objectMapper objectMapper
      * @param publicationService    publicationService
      * @param environment  environment
      */
-    public FetchPublicationHandler(ObjectMapper objectMapper, PublicationService publicationService,
+    public FetchPublicationHandler(PublicationService publicationService,
+                                   ObjectMapper objectMapper,
                                    Environment environment) {
-        super(objectMapper, environment);
+        super(Void.class, environment);
+        this.objectMapper = objectMapper;
         this.publicationService = publicationService;
     }
 
     @Override
-    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
-        String authorization;
-        UUID identifier;
-        try {
-            JsonNode event = objectMapper.readTree(input);
-            authorization = Optional.ofNullable(event.at(HEADERS_AUTHORIZATION).textValue())
-                    .orElseThrow(() -> new IllegalArgumentException(MISSING_AUTHORIZATION_IN_HEADERS));
-            identifier = UUID.fromString(Optional.ofNullable(event.at(PATH_PARAMETERS_IDENTIFIER).textValue())
-                    .orElseThrow(() -> new IllegalArgumentException(MISSING_IDENTIFIER_IN_PATH_PARAMETERS)));
-        } catch (Exception e) {
-            logError(e);
-            writeErrorResponse(output, BAD_REQUEST, e);
-            return;
-        }
-
-        log("Request for identifier: " + identifier.toString());
+    protected JsonNode processInput(Void input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        Publication publication = publicationService.getPublication(
+                RequestUtil.getIdentifier(requestInfo),
+                RequestUtil.getAuthorization(requestInfo));
+
+        JsonNode publicationJson = objectMapper.valueToTree(publication);
+        addContext(publicationJson);
```

Comment:
Refactor these lines in one method, for better readability and avoidance of side-effects